core/src/main/java/com/segment/analytics/kotlin/core/Analytics.kt

```diff
@@ -527,6 +533,10 @@ open class Analytics protected constructor(
      * user logs out
      */
     fun reset() {
+        userInfo.userId = null
```

Comment:
wonder if we could just new a `UserInfo` object instead of doing assignments individually?